### PR TITLE
Add Automated Coverity Scanning

### PR DESCRIPTION
**Pull request by jadenabrams100**
This PR adds automatic Coverity functionality to the GRASS GIS repository. This action will (should) tell the Synopsys Coverity action to run the scan, with the results being posted on the Coverity Website. The following things will be needed in secrets for the action to work:

* COVERITY_URL: the URL for Coverity
* COVERITY_USER: the Coverity username
* COVERITY_PASSPHRASE: the Coverity passphrase

This action is based off of https://sig-product-docs.synopsys.com/bundle/bridge/page/documentation/c_github-coverity.html and will be worked on with @wenzeslaus

### CODE REVIEW

**Comment by neteler** [comment]
(if needed, I have the credentials to our existing Coverity profile)

**Comment by wenzeslaus** [comment]
Thanks @neteler. I checked and I see at least one token, so it looks like I can handle that.

**Comment by echoix** [comment]
Can we (I) have a little explanation/pitch on what Coverity does and brings us? I see it's a commercial SaaS, but probably free for us, but what are its strengths?

**Comment by jadenabrams100** [comment]
> Can we (I) have a little explanation/pitch on what Coverity does and brings us? I see it's a commercial SaaS, but probably free for us, but what are its strengths?

Coverity is an enterprise code scanning tool that is also free for open source projects. One of the suggestions for our team was to add integration to Coverity. Since grass is already registered to 
 [https://scan.coverity.com/projects/grass](coverity), this action would make it so these scans would be run automatically. The advantage of this tool is that it provides another powerful way for us to detect vulnerabilities, in particular for C code.

**Comment by echoix** [comment]
> > Can we (I) have a little explanation/pitch on what Coverity does and brings us? I see it's a commercial SaaS, but probably free for us, but what are its strengths?
> 
> 
> 
> Coverity is an enterprise code scanning tool that is also free for open source projects. One of the suggestions for our team was to add integration to Coverity. Since grass is already registered to 
> 
>  [https://scan.coverity.com/projects/grass](coverity), this action would make it so these scans would be run automatically. The advantage of this tool is that it provides another powerful way for us to detect vulnerabilities, in particular for C code.

When you say _for your team_, what are you referring to?

File: .github/workflows/coverity.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Coverity Scan
+# Created using https://sig-product-docs.synopsys.com/bundle/bridge/page/documentation/c_github-coverity.html as a template
+# Author: Jaden Abrams <jadenabrams100@gmail.com>
```
[the remark this answers is not in the document]
Comment (jadenabrams100) [reply .github/workflows/coverity.yml:3]: I wasn't entirely sure how the attribution for new files work, so I added that in at the top like how it is for source files.

File: .github/workflows/coverity.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+on:
+  workflow_dispatch: # run whenever a contributor calls it
+  schedule: 
+    - cron: '48 5 * * *' # Run at 05:48
+
```
[the remark this answers is not in the document]
Comment (jadenabrams100) [reply .github/workflows/coverity.yml:8]: It is, Coverity will only let GRASS run the scan a maximum of twice per day based on its size. https://scan.coverity.com/faq#frequency

File: .github/workflows/coverity.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    steps:
+      - name: Checkout Source
+        uses: actions/checkout@v3
+      - name: Coverity Scan
+        uses: synopsys-sig/synopsys-action@v1.6.0
+        with:
+          ### SCANNING: Required fields
+          coverity_url: ${{ secrets.COVERITY_URL }}               # The URL to Coverity
+          coverity_user: ${{ secrets.COVERITY_USER }}             # The user for the Coverity project
+          coverity_passphrase: ${{ secrets.COVERITY_PASSPHRASE }} # The password for the Coverity user
```
[the remark this answers is not in the document]
Comment (wenzeslaus) [reply .github/workflows/coverity.yml:21]: Yes.

File: .github/workflows/coverity.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (echoix) [reply .github/workflows/coverity.yml]: Makes sense!

**Review by echoix — APPROVED** [review]

**Comment by petrasovaa** [comment]
> When you say _for your team_, what are you referring to?
See https://grasswiki.osgeo.org/wiki/NSF_POSE_Project_2023-2025_Timeline#Code_Security

**Review by wenzeslaus — APPROVED** [review]
We need to test this by merging, but it looks hopeful. I checked how things look in Coverity and it seems to be just waiting for another upload. With help of [some alternative documentation](https://community.synopsys.com/s/article/Coverity-Integrations-GitHub-with-GitHub-Hosted-Runners), I created the three secrets. The user is my user and pass phrase is project token (the doc suggests it may work). Let's stay in touch. Thanks!

**Comment by wenzeslaus** [comment]
Also, I added @jadenabrams100 to as a Contributor to grass project on Coverity (for the duration of his team's class project) to have direct access to the issues and I invited @echoix.

**Comment by echoix** [comment]
> Also, I added @jadenabrams100 to as a Contributor to grass project on Coverity (for the duration of his team's class project) to have direct access to the issues and I invited @echoix.

Perfect! I just registered, will take a look in the near future

**Comment by wenzeslaus** [comment]
The job fails because of the URL. I have no idea what should be there. The outdated example nor any of the URLs I got from the examples in the project page at Coverity Scan worked.

**Comment by echoix** [comment]
The action has a recent update that Renovate did a PR for. Didn't get through yet.

**Pull request merged**